sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonSquidSensor.java

```diff
@@ -61,58 +59,57 @@
   private static final Number[] FUNCTIONS_DISTRIB_BOTTOM_LIMITS = {1, 2, 4, 6, 8, 10, 12, 20, 30};
   private static final Number[] FILES_DISTRIB_BOTTOM_LIMITS = {0, 5, 10, 20, 30, 60, 90};
 
-  private final AnnotationCheckFactory annotationCheckFactory;
+  private final Checks<SquidAstVisitor<Grammar>> checks;
   private final FileLinesContextFactory fileLinesContextFactory;
 
-  private Project project;
   private SensorContext context;
   private AstScanner<Grammar> scanner;
-  private ModuleFileSystem fileSystem;
+  private FileSystem fileSystem;
   private ResourcePerspectives resourcePerspectives;
 
-  public PythonSquidSensor(RulesProfile profile, FileLinesContextFactory fileLinesContextFactory, ModuleFileSystem fileSystem, ResourcePerspectives resourcePerspectives) {
-    this.annotationCheckFactory = AnnotationCheckFactory.create(profile, CheckList.REPOSITORY_KEY, CheckList.getChecks());
+  public PythonSquidSensor(FileLinesContextFactory fileLinesContextFactory, FileSystem fileSystem, ResourcePerspectives perspectives, CheckFactory checkFactory) {
+    this.checks = checkFactory
+        .<SquidAstVisitor<Grammar>>create(CheckList.REPOSITORY_KEY)
+        .addAnnotatedChecks(CheckList.getChecks());
     this.fileLinesContextFactory = fileLinesContextFactory;
     this.fileSystem = fileSystem;
-    this.resourcePerspectives = resourcePerspectives;
+    this.resourcePerspectives = perspectives;
   }
 
   public boolean shouldExecuteOnProject(Project project) {
-    return !fileSystem.files(FileQuery.onSource().onLanguage(Python.KEY)).isEmpty();
+    return fileSystem.hasFiles(fileSystem.predicates().hasLanguage(Python.KEY));
```

Comment:
The new code is not equivalent to the old one: FileQuery.onSource() should be transformed in a predicate on file type.